src/App/Components/DisplayArticle/Ratings.js

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable react/no-unused-state */
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { ToastContainer, toast } from 'react-toastify';
+import propTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import Rater from 'react-star-ratings';
+import Rating from './Rating';
+import actions from '../../../Redux/Actions/articleRatings';
+
+const { getRatings, rate } = actions;
+
+export class Ratings extends Component {
+  static propTypes = {
+    getRatings: propTypes.func,
+    ratings: propTypes.object,
+    errorMessage: propTypes.string,
+    successMessage: propTypes.string,
+    rate: propTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      average: undefined,
+      allUsers: undefined,
+      stats: [],
+      rating: 0,
+      redirect: false,
+      areRatingsShown: false,
+      isLoggedIn: false
+    };
+  }
+
+  componentDidMount() {
+    const { getRatings: getArticleRatings } = this.props;
+    getArticleRatings('What-is-a-Version-1-UUID');
+  }
+
+  componentWillReceiveProps({ ratings, errorMessage, successMessage }) {
+    if (!isEmpty(ratings)) {
+      this.setState({
+        average: ratings.average.toString().substring(0, 3),
+        allUsers: ratings.data.allUsers,
+        stats: ratings.data.statistics
+      });
+    }
+
+    if (errorMessage) toast.error(errorMessage);
+    if (successMessage) toast.success(successMessage);
+  }
+
+  rate = async (rating) => {
+    const redirectToLogin = () => this.setState({ redirect: true });
+    const Toast = ({ closeToast }) => (
+      <div>
+        <p className="ogin-popup-message">Please, login first...</p>
+        <p className="login-popup-buttons">
+          <button type="submit" onClick={redirectToLogin}>
+            Yes
+          </button>
+          <button type="submit" onClick={closeToast}>
+            No
+          </button>
+        </p>
+      </div>
+    );
+
+    const token = localStorage.getItem('token');
+    if (token) {
+      this.setState({ isLoggedIn: true });
+      const {
+        props: { rate: postRating, getRatings: getArticleRatings },
+        state: { rating: userRating }
+      } = this;
+      if (rating !== userRating) {
+        this.setState({ rating });
+        await postRating(rating, 'What-is-a-Version-1-UUID', token);
+        getArticleRatings('What-is-a-Version-1-UUID');
```

Comment:
same here @nkpremices, if you are using deferent component, you may get `slug` from the parent component

Reply:
Same here @Ramadhan0